airbyte-integrations/connectors/source-recharge/source_recharge/manifest.yaml

```diff
@@ -0,0 +1,269 @@
+      Products endpoint has 422 error with 2021-11 API version
+    $ref: "#/definitions/base_deprecated_api_stream"
+    $parameters:
+      name: "products"
+      data_path: "products"
+  # SHOP
+  shop_stream:
+    description: >-
+      Shop Stream: https://developer.rechargepayments.com/v1-shopify?python#shop
+      Shop endpoint is not available in 2021-11 API version
+    $ref: "#/definitions/base_deprecated_api_stream"
+    retriever:
+      $ref: "#/definitions/retriever_api_deprecated"
+      record_selector:
+        $ref: "#/definitions/selector"
+        extractor:
+          type: DpathExtractor
+          field_path: []
+    primary_key: ["shop", "store"]
+    $parameters:
+      name: "shop"
+
+  # INCREMENTAL STREAMS
+  # ADDRESSES
+  addresses_stream:
+    description: >-
+      Addresses Stream: https://developer.rechargepayments.com/2021-11/addresses/list_addresses
+    $ref: "#/definitions/base_incremental_stream"
+    $parameters:
+      name: "addresses"
+      data_path: "addresses"
+  # CHARGES
+  charges_stream:
+    description: >-
+      Charges Stream: https://developer.rechargepayments.com/2021-11/charges/charge_list
+    $ref: "#/definitions/base_incremental_stream"
+    $parameters:
+      name: "charges"
+      data_path: "charges"
+  # CUSTOMERS
+  customers_stream:
+    description: >-
+      Customers Stream: https://developer.rechargepayments.com/2021-11/customers/customers_list
+    $ref: "#/definitions/base_incremental_stream"
+    $parameters:
+      name: "customers"
+      data_path: "customers"
+  # DISCOUNTS
+  discounts_stream:
+    description: >-
+      Discounts Stream: https://developer.rechargepayments.com/2021-11/discounts/discounts_list
+    $ref: "#/definitions/base_incremental_stream"
+    $parameters:
+      name: "discounts"
+      data_path: "discounts"
+  # ONETIMES
+  onetimes_stream:
+    description: >-
+      Onetimes Stream: https://developer.rechargepayments.com/2021-11/onetimes/onetimes_list
+    $ref: "#/definitions/base_incremental_stream"
+    $parameters:
+      name: "onetimes"
+      data_path: "onetimes"
+  # SUBSCRIPTIONS
+  subscriptions_stream:
+    # description: >-
+    #   Subscriptions Stream: https://developer.rechargepayments.com/2021-11/subscriptions/subscriptions_list
+    $ref: "#/definitions/base_incremental_stream"
+    $parameters:
+      name: "subscriptions"
+      data_path: "subscriptions"
+
+streams:
+  - "#/definitions/addresses_stream"
+  - "#/definitions/charges_stream"
+  - "#/definitions/collections_stream"
+  - "#/definitions/customers_stream"
+  - "#/definitions/discounts_stream"
+  - "#/definitions/metafields_stream"
+  - "#/definitions/onetimes_stream"
+  - "#/definitions/products_stream"
+  - "#/definitions/shop_stream"
+  - "#/definitions/subscriptions_stream"
+  # The `orders` stream remains implemented in `streams.py` due to:
+  # 1. Inability to resolve `$ref` conditionally
```

Comment:
We discussed this during a meeting but could you add more information as to which behavior would require this to help the reader know where to look for?

Reply:
No problem,

Different API versions require different Pagination approach.

1) we have `modern` API == 2021-11, which uses `cursor` pagination
2) we have `deprecated` API == 2021-01^ which uses `page increment` pagination

Both of these have different inputs and outputs, thus requiring to be switched based on the `config["use_orders_deprecated_api"]`.

If  `config["use_orders_deprecated_api"] == True, the 2021-01 version should be used.
if `config["use_orders_deprecated_api"] == False, the 2021-11 version should be used.

Given the info above:
- there was a chance to dynamically switch between Paginators, that are already predefined, based on the `config["use_orders_deprecated_api"] value and applied to the `SimpleRetriever` as the pagination to go with.

Then in an ideal world, the `orders` stream migrated would look like:
```yaml
# PAGINATORS
paginator_A:
   type: PageIncrement
    .....

paginator_B:
   type: Cursor
   ....

# STREAMS
orders:
   primary_key: "blaBla",
   .....
   retriever:
    $ref: "#/definitions/some_base_retriever"
    paginator:
        $ref: "{{ '#/definitions/paginator_A' if config["use_orders_deprecated_api"] else '#/definitions/paginator_B' }}"
```